src/Network/Email/Email.php

```diff
@@ -1885,4 +1890,73 @@ protected function _getContentTypeCharset()
         }
         return strtoupper($this->charset);
     }
+
+    /**
+     * Serializes the email object to a value that can be natively serialized and re-used
+     * to clone this email instance.
+     *
+     * It has certain limitations for viewVars that are good to know:
+     *
+     *    - ORM\Query executed and stored as resultset
+     *    - File resources read and stored as contents
+     *    - SimpleXmlElements stored as associative array
+     *    - Closures, Exceptions, PDOs and SimpleXmlElements can NOT be handled.
+     *
+     * @return array Serializable array of configuration properties.
+     * @throws \Exception When a view var object can not be properly serialized.
+     */
+    public function jsonSerialize()
+    {
+        $properties = [
+            '_to', '_from', '_replyTo', '_cc', '_bcc', '_subject', '_returnPath', '_readReceipt',
+            '_template', '_layout', '_viewRender', '_viewVars', '_theme', '_helpers', '_emailFormat',
+            '_attachments', '_domain', '_messageId', '_headers', 'charset', 'headerCharset',
+        ];
+
+        $array = [];
+
+        foreach ($properties as $property) {
+            $array[$property] = $this->{$property};
+        }
+
+        // Iterate through view vars to clean up and normalize
+        array_walk_recursive($array['_viewVars'], function (&$item) {
+            // Store file contents instead of resource.
+            if (is_resource($item) && get_resource_type($item) == 'stream') {
+                $item = stream_get_contents($item);
+            }
+
+            if ($item instanceof Exception) {
+                $item = (string)$item;
+            }
+
+            // Break on stuff that can't be serialized
+            if (
+                $item instanceof Closure ||
+                $item instanceof PDO
+            ) {
+                // FIXME better message? another exception?
+                throw new NotImplementedException(get_class($item) . ' serialization');
```

Comment:
I would throw just raise `\RuntimeException`. If you accept the `$key` parameter into your closure you can indicate which view variable caused the issue as well.